src/ComposerConstraint.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\lightning_dev;
+
+class ComposerConstraint {
+
+  /**
+   * E.g. '^2.8 || ^3.0'.
+   *
+   * @var string
+   */
+  private $constraint = '';
+
+  /**
+   * @param string $constraint
+   */
+  public function __construct($constraint) {
+    $this->constraint = $constraint;
+  }
+
+  /**
+   * @return string[]
+   *   E.g. ['^2.8', '^3.0'].
+   */
+  public function getRanges() {
+    preg_match_all('/[0-9a-zA-Z\~\>\=\-\<\.\^\*]+/', $this->constraint, $matches);
+
+    return $matches[0];
+  }
+
+  /**
+   * @return string
+   *   E.g. '8.4.x-dev || 8.5.x-dev'.
+   */
+  public function getCoreDev() {
+    return $this->getDev([static::class, 'coreRangeToDev']);
+  }
+
+  /**
+   * @return string
+   *   E.g. '2.x-dev || 3.x-dev'.
+   */
+  public function getLightningDev() {
+    return $this->getDev([static::class, 'lightningRangeToDev']);
+  }
+
+  /**
+   * @param callable $callback
+   *
+   * @return string
+   */
+  private function getDev($callback) {
```

Comment:
$callback should be type hinted as `callable`. :)

Reply:
Yeah, for some reason I thought that was only available in PHP 7.